p2p/server.go

```diff
@@ -911,7 +911,9 @@ func (srv *Server) listenLoop() {
 			fd = newMeteredConn(fd, true, addr)
 			srv.log.Trace("Accepted connection", "addr", fd.RemoteAddr())
 		}
+		srv.loopWG.Add(1)
```

Comment:
Adding to loopWG from within the loop is unsafe because it can lead to Add being called concurrently with Wait. It would be better to wait for the slots channel on shutdown in the loop.

Reply:
Ok, I've squashed in an alternate solution, PTAL